server/src/main/java/org/opensearch/action/search/SearchRequestStats.java

```diff
@@ -73,20 +73,29 @@ public long getTookMetric() {
 
     @Override
     protected void onPhaseStart(SearchPhaseContext context) {
```

Comment:
Wondering if we can do the try/catch here instead. `SearchPhaseName` already contains the exact list of phases needed. Then we don't need to add a catchall `SearchPhaseName` and don't need the `shouldTrack` attribute.
```
    protected void onPhaseStart(SearchPhaseContext context) {
        try {
            phaseStatsMap.get(context.getCurrentPhase().getSearchPhaseName()).current.inc();
        } catch (IllegalArgumentException e) {
            // Do nothing if phase is not a member of SearchPhaseName enum
        }
    }
```

Reply:
This is much cleaner, good idea. I've made this change.